grakn-engine/src/main/java/ai/grakn/engine/tasks/connection/RedisConnection.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.tasks.connection;
+
+import ai.grakn.concept.ConceptId;
+import ai.grakn.engine.GraknEngineConfig;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+
+import java.util.function.Function;
+
+/**
+ * <p>
+ *     Connection To Redis Server
+ * </p>
+ *
+ * <p>
+ *    A class which manages the connection to the central Redis cache.
+ *    This serves as a cache for keeping track of the counts of concepts which may be in need of sharding.
+ * </p>
+ *
+ * @author fppt
+ */
+public class RedisConnection {
+    private static final GraknEngineConfig config = GraknEngineConfig.getInstance();
+    private static RedisConnection redis;
+
+    private JedisPool jedisPool;
+
+    private RedisConnection(){
+        JedisPoolConfig poolConfig = new JedisPoolConfig();
+        poolConfig.setMaxTotal(128);
+        jedisPool = new JedisPool(poolConfig, config.getProperty(GraknEngineConfig.REDIS_SERVER_URL),
+                config.getPropertyAsInt(GraknEngineConfig.REDIS_SERVER_PORT));
+    }
+
+    /**
+     * Returns and possibly initliases a connection pool to the redis server.
+     *
+     * @return a connction to the redis server.
+     */
+    public static synchronized RedisConnection getConnection(){
+        if(redis == null) redis = new RedisConnection();
+        return redis;
+    }
```

Comment:
This is `synchronized`, but does it definitely do the right thing? This will mean only one thing can establish a connection at once, but it is still possible for two things to call `adjustCount` simultaneously.